components/layout_2020/intrinsic.rs

```diff
@@ -0,0 +1,69 @@
+use crate::style_ext::ComputedValuesExt;
+use style::Zero;
+use style::properties::ComputedValues;
+use style::values::computed::{Length, LengthPercentage};
+
+#[derive(Debug, Clone)]
+pub(crate) struct IntrinsicSizes {
+    pub min_content: Length,
+    pub max_content: Length,
+}
+
+impl IntrinsicSizes {
+    pub fn zero() -> Self {
+        Self {
+            min_content: Length::zero(),
+            max_content: Length::zero(),
+        }
+    }
+}
+
+/// https://dbaron.org/css/intrinsic/#outer-intrinsic
+pub(crate) fn outer_intrinsic_inline_sizes(
+    style: &ComputedValues,
+    get_inner_intrinsic_sizes: &dyn Fn() -> IntrinsicSizes,
+) -> IntrinsicSizes {
+    // FIXME: account for 'min-width', 'max-width', 'box-sizing'
+
+    let specified = style.box_size().inline;
+    // Percentages for 'width' are treated as 'auto'
+    let specified = specified.map(|lp| lp.as_length());
+    // The (inner) min/max-content are only used for 'auto'
+    let mut outer = match specified.non_auto().flatten() {
+        None => get_inner_intrinsic_sizes(),
+        Some(length) => IntrinsicSizes {
+            min_content: length,
+            max_content: length,
+        }
+    };
```

Comment:
Oh yeah I totally used the wrong name here. `computed` is correct per https://drafts.csswg.org/css-cascade/#computed (though it doesn’t match the `getComputedStyle` API, but it’s the API which is named incorrectly) for the first binding.

I’ve renamed to `inline_size`, but kept the bindings so that comments apply to each step and in the expected order.